File: Adhiti_Shaili/Day 12/Min XOR value.cpp
#include<bits/stdc++.h>
using namespace std;

int Solution::findMinXor(vector<int> &A) {
    int maxi=INT_MAX;
    sort(A.begin(),A.end());
    for(int i=0;i<A.size()-1;i++)
    {
        maxi=min(maxi,A[i]^A[i+1]);
    }
    return maxi;
}

File: sanket-dalvi/Day10/minXorValue.py
class Solution:
    # @param A : list of integers
    # @return an integer
    def findMinXor(self, A):
        A.sort()
        return min(map(lambda i: A[i-1]^A[i], range(1, len(A))))

# Test
print(Solution().findMinXor([12,4,6,2]))